.github/workflows/ci.yml

```diff
@@ -24,18 +23,45 @@ jobs:
         with:
           node-version: ${{ matrix.node-version }}
 
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - name: Check yarn cache
+        uses: actions/cache@v2
+        id: cache-yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Check node_modules cache
+        id: cache-node-modules
+        uses: actions/cache@v2
+        with:
+          path: |
+            node_modules
+            */*/node_modules
+          key: ${{ runner.os }}-${{ matrix.node-version }}-nodemodules-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ matrix.node-version }}-nodemodules-
+
       - name: Install packages
         run: npm i
 
-      - name: Build contracts
+      - name: Prepare environment
+        run: cp .env.example .env
+
+      - name: Build contract
         run: npx hardhat compile
 
       - name: Test contract
         run: npx hardhat test
         env:
           POLYGON_URL: ${{ secrets.POLYGON_URL }}
-          POLYGON_PRIVATE_KEY: ${{ secrets.POLYGON_PRIVATE_KEY }}
```

Comment:
What was the issue here?

Reply:
@marceljay are you asking why was the key removed? If so, it is now reused from `.env.example` which stores the default hardhat key.

Reply:
Ah yeah sure it's only for testing.